cmd/env.go

```diff
@@ -1,61 +1,48 @@
 package cmd
 
 import (
-	"context"
-	"fmt"
-
 	"github.com/celestiaorg/celestia-node/node"
+
+	"context"
 )
 
-// Env is an environment for CLI commands.
-// It can be used to:
-// 1. Propagate values from parent to child commands.
-// 2. To group common logic that multiple commands rely on.
-// Usage can be extended.
-// TODO(@Wondertan): We should move to using context only instead.
-//  Env, in fact, only keeps some additional fields which should be
-//  kept in the context directly using WithValue (#965)
-type Env struct {
-	NodeType  node.Type
-	StorePath string
-
-	opts []node.Option
+// NodeType reads the node.Type from the context.
+func NodeType(ctx context.Context) node.Type {
+	return ctx.Value(nodeTypeKey{}).(node.Type)
```

Comment:
This could panic if the values are not given, which is fine(?). Yeah, it's fine because if the programmer forgot to pass the Node type, that's his fault, and panic will show during testing. Although some would consider erroring a better thing

Reply:
Agreed, resolving